internal/driver/discoverymode_test.go

```diff
@@ -0,0 +1,115 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestIsValid verifies discovery mode setting.
+func TestIsValid(t *testing.T) {
+
+	tests := []struct {
+		input    DiscoveryMode
+		expected bool
+	}{
+		{
+			input:    ModeNetScan,
+			expected: true,
+		},
+		{
+			input:    ModeMulticast,
+			expected: true,
+		},
+		{
+			input:    ModeBoth,
+			expected: true,
+		},
+		{
+			input:    "invalidValue",
+			expected: false,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(string(test.input), func(t *testing.T) {
+			result := test.input.IsValid()
+			assert.Equal(t, test.expected, result)
+		})
+	}
+}
+
+// TestIsMulticastEnabled verifies multicast setting.
+func TestIsMulticastEnabled(t *testing.T) {
+
+	tests := []struct {
+		input    DiscoveryMode
+		expected bool
+	}{
```

Comment:
these two test functions can be combined if using something like this:
also, i recommend to rename `input` to `mode`
```suggestion
	tests := []struct {
		mode    DiscoveryMode
		multicast bool
		netscan bool
	}{
```